dash/_utils.py

```diff
@@ -57,17 +58,31 @@ def get_asset_path(requests_pathname, asset_path, asset_url_path):
 def get_relative_path(requests_pathname, path):
     if requests_pathname == '/' and path == '':
         return '/'
-    if requests_pathname == '/' and not path.startswith('/'):
-        return path
+    elif requests_pathname != '/' and path == '':
+        return requests_pathname
     elif not path.startswith('/'):
-        return requests_pathname + path
+        raise exceptions.UnsupportedRelativePath(
+            "Paths that aren't prefixed with a leading / are not supported.\n" +
+            "You supplied: {}".format(path)
+        )
     return "/".join(
         [
             requests_pathname.rstrip("/"),
             path.lstrip("/")
         ]
     )
 
+def strip_relative_path(requests_pathname, path):
+    if path is None:
+        return None
+    elif not path.startswith('/'):
+        raise exceptions.UnsupportedRelativePath(
+            "Paths that aren't prefixed with a leading / are not supported.\n" +
+            "You supplied: {}".format(path)
+        )
+    if requests_pathname != '/':
+        path = path.replace(requests_pathname, '')
```

Comment:
Is there any reason we'd get here with `path` not starting with  `requests_pathname`? Should it be an error if we do?

And regardless of that, we should only remove `requests_pathname` at the beginning of `path`, right? I mean, you could imagine an app hosted at `app/` with a path `app/sub-app/` or whatever.

Reply:
For the `dcc.Location.pathname` usage, I can't imagine a case where we'd get a path that doesn't start with ` requests_pathname`. We could error out.

Good point re beginning of path. The only exception is if the prefix is `/my-dash-app/` but then the app might be served under `/my-dash-app` and `dcc.Location` might pass `/my-dash-app` through. In Dash Enterprise, we redirect to the trailing slash version (e.g. https://dash-demo.plotly.host/ddk-oil-and-gas-demo redirects to https://dash-demo.plotly.host/ddk-oil-and-gas-demo/) but I could imagine that that redirect logic could inadvertently change in the future sometime.

I updated the logic and added a test case for this in 685cd69